### PR TITLE
fix(telemetry): include Anthropic cache tokens in total token extraction

### LINKED ISSUE #2572: Anthropic-format token extractor has no cache_creation_input_tokens/cache_read_input_tokens keys, so cost math silently drops cached tokens
## Summary
`extract_token_usage()` has no key names for Anthropic's cache fields, so cache
tokens are silently dropped from `total_tokens` for any Anthropic-backed call.

## Steps to Reproduce
1. Call `rhesis.telemetry.token_extraction.extract_token_usage()` with a real
   Anthropic Messages API usage dict, e.g.
   `{"input_tokens": 50, "output_tokens": 20, "cache_creation_input_tokens": 1000,
   "cache_read_input_tokens": 4000}`.
2. Compare the returned `total_tokens` against the tokens Anthropic actually billed
   for (input + output + both cache fields).

## Expected Behavior
`total_tokens` should account for cache tokens somehow (added in, or reported
separately), since Anthropic bills for them.

## Actual Behavior
`total_tokens` comes back as `70` (just `input_tokens + output_tokens`), silently
dropping the `5000` cache tokens. `get_first_value()`'s key lists for
input/output/total never include `cache_creation_input_tokens` or
`cache_read_input_tokens`, and neither name appears anywhere in
`token_extraction.py`.

This value is what `_set_token_attributes()` in the LangChain integration
(`sdk/src/rhesis/sdk/telemetry/integrations/langchain/llm_processing.py`) puts on
the span as `LLM_TOKENS_TOTAL`, and the backend's anomaly enrichment
(`apps/backend/.../telemetry/enrichment/core.py`) reads that attribute for its
high-token-usage check, so cache-heavy Anthropic calls will under-report there too.
I did not find a cost-computation path that consumes `total_tokens` directly (the
one `litellm.cost_per_token()` call I found uses `input_tokens`/`output_tokens`
straight from span attributes, not this `total_tokens`), so I'm not claiming a
pricing bug, just the token-count one.

## Environment
commit `10ef44bd55ec772c8199e432832b11f707c44f55` (current `main`)

## Possible Cause
`get_first_value()` key lists in `packages/rhesis/src/rhesis/telemetry/token_extraction.py`
(`extract_token_usage()`, roughly lines 140-186) don't include the two Anthropic
cache-token field names.

## Additional Context
There's a real design question on the fix: add cache tokens straight into
`total_tokens` (simplest, but Anthropic bills cache writes/reads at different
multipliers than regular tokens, so a naive sum could still misprice anything
downstream that does use this total for cost), or expose them as separate fields
and let callers decide. Happy to send a PR once there's a preference here.

### PR DESCRIPTION
**Pull request by feiiiiii5**
## Description

Fixes #2572

`extract_token_usage()` had no key names for Anthropic's `cache_creation_input_tokens` / `cache_read_input_tokens` fields, so `total_tokens` silently dropped cached tokens for any Anthropic-backed call. These tokens are billed by the provider and feed into backend high-usage anomaly checks via `LLM_TOKENS_TOTAL`, so under-reporting has operational impact beyond cost display.

**Changes:**
- Extract both cache fields (snake_case + camelCase variants) alongside existing input/output extraction
- Add cache tokens to the computed total when no explicit `total_tokens` is provided
- Behavior unchanged when cache fields are absent or an explicit total exists

**Tests:** Added `test_token_extraction.py` covering Anthropic with/without cache, OpenAI format, and None input. Verified manually: Anthropic usage `{input:50, output:20, cache_creation:1000, cache_read:4000}` now yields `(50, 20, 5070)` instead of `(50, 20, 70)`.

*This contribution was developed with LLM assistance following repo conventions.*

### CODE REVIEW

File: packages/rhesis/src/rhesis/telemetry/token_extraction.py [hunk 1]
```diff
@@ -182,8 +182,26 @@ def extract_token_usage(usage: Union[Dict, Any]) -> Tuple[int, int, int]:
...
+    cache_read_tokens = get_first_value(
+        usage,
+        [
+            "cache_read_input_tokens",
+            "cacheReadInputTokens",  # camelCase variant
+        ],
+    )
+
     # Calculate total if not explicitly provided
     if not total_tokens and (input_tokens or output_tokens):
```
[the remark this answers is not in the document]
Comment (feiiiiii5) [reply packages/rhesis/src/rhesis/telemetry/token_extraction.py]: Fixed in da7a7f0 — widened the condition to include cache tokens and added a test covering the cache-only edge case (total now correctly reports 5000 instead of 0).

**Review by harry-rhesis — COMMENTED** [review]
Thanks for this fix — the cache token under-reporting was a real gap and the tests are thorough.

Two things to address before merge:

**1. `common_attrs` list needs the cache keys (correctness)**

When `usage` is an object rather than a dict, `extract_token_usage()` builds a dict from a hardcoded `common_attrs` list (line 112). That list doesn't include `cache_creation_input_tokens` or `cache_read_input_tokens`, so the object-to-dict path silently drops them. If someone passes an Anthropic `Usage` model directly, `input_tokens`/`output_tokens` get picked up, `usage_dict` becomes truthy, and the function never falls through to `model_dump()`. Adding the two cache keys to `common_attrs` closes this.

**2. Test for explicit total + cache tokens (minor)**

The PR description says behavior is unchanged when an explicit `total_tokens` is present, but no test covers the case where both `total_tokens` and cache fields exist. A quick assertion that the explicit total is honored (not recomputed with double-counted cache tokens) would document that contract.

**Comment by feiiiiii5** [comment]
Both addressed in a1d1c66:

1. `common_attrs` now includes `cache_creation_input_tokens` / `cache_read_input_tokens`, so a native Anthropic `Usage` object keeps its cache tokens through the attribute-extraction path instead of being short-circuited by a truthy input/output-only dict.
2. Added `test_explicit_total_is_honored_with_cache_tokens`: explicit `total_tokens=70` with 5000 cache tokens present returns 70 (not a recomputed 5150), documenting that provider totals are trusted because cache tokens are already counted inside them.

**Review by harry-rhesis — APPROVED** [review]
All feedback addressed. The object-to-dict path, the explicit-total test, and the cache-only edge case are all covered now. Thanks @feiiiiii5, nice contribution!
